crates/ruff_cli/src/diagnostics.rs

```diff
@@ -438,21 +437,23 @@ pub(crate) fn lint_stdin(
     noqa: flags::Noqa,
     autofix: flags::FixMode,
 ) -> Result<Diagnostics> {
-    // Extract the sources from the file.
-    let LintSources {
-        source_type,
-        source_kind,
-    } = match LintSources::try_from_source_code(contents, path) {
-        Ok(sources) => sources,
-        Err(SourceExtractionError::Io(err)) => {
-            // SAFETY: An `io::Error` can only occur if we're reading from a path.
-            return Ok(Diagnostics::from_io_error(&err, path.unwrap(), settings));
-        }
-        Err(SourceExtractionError::Diagnostics(diagnostics)) => {
-            return Ok(*diagnostics);
-        }
+    let SourceType::Python(source_type) = path.map(SourceType::from).unwrap_or_default() else {
+        return Ok(Diagnostics::default());
     };
```

Comment:
This is the kind of thing I'm trying to prevent with this PR. Previously, this code ignored `pyproject.toml` entirely and would've ended up trying to parse `pyproject.toml` files as Python files... Now, we're at least forced to ignore them.